plugins/net_plugin/net_plugin.cpp

```diff
@@ -1740,25 +1745,38 @@ namespace eosio {
       } FC_LOG_AND_DROP();
       if( sb ) {
          // Skip transmitting block this loop if threshold exceeded
-         if( block_sync_rate_limit > 0 && peer_syncing_from_us ) {
-            auto elapsed = std::chrono::duration_cast<std::chrono::seconds>(get_time() - connection_start_time);
-            auto current_rate = double(block_sync_bytes_sent) / elapsed.count();
+         if (block_sync_send_start == 0ns) { // start of enqueue blocks
+            block_sync_send_start = get_time();
+            block_sync_send_bytes_sent = 0;
+         }
+         if( block_sync_rate_limit > 0 && block_sync_send_bytes_sent > 0 && peer_syncing_from_us ) {
+            auto now = get_time();
+            auto elapsed_us = std::chrono::duration_cast<std::chrono::microseconds>(now - block_sync_send_start);
+            double current_rate = (double(block_sync_bytes_sent) / elapsed_us.count()) * 100000;
```

Comment:
If you are changing variables names from my comment above, please make this line a bit more readable by renaming `current_rate` to  `current_rate_sec` or something like that. it will make it obvious that these are seconds and why you divide by `100'000`. This is optional in case you are changing the code.